addons/web/models/ir_qweb.py

```diff
@@ -45,17 +46,26 @@ def record_to_html(self, record, field_name, options):
         sha = hashlib.sha1(str(getattr(record, '__last_update')).encode('utf-8')).hexdigest()[0:7]
         max_size = '' if max_size is None else '/%s' % max_size
         avoid_if_small = '&avoid_if_small=true' if options.get('avoid_if_small') else ''
-        src = '/web/image/%s/%s/%s%s?unique=%s%s' % (record._name, record.id, options.get('preview_image', field_name), max_size, sha, avoid_if_small)
 
-        alt = None
+        if options.get('filename-field') and getattr(record, options['filename-field'], None):
+            filename = record[options['filename-field']]
+        elif options.get('filename'):
+            filename = options['filename']
+        else:
+            filename = record.display_name
+
+        src = '/web/image/%s/%s/%s%s/%s?unique=%s%s' % (record._name, record.id, options.get('preview_image', field_name), max_size, url_quote(filename), sha, avoid_if_small)
+
         if options.get('alt-field') and getattr(record, options['alt-field'], None):
             alt = escape(record[options['alt-field']])
         elif options.get('alt'):
             alt = options['alt']
+        else:
+            alt = escape(record.display_name)
```

Comment:
are you sure that it will work if simple quote in diplay name???
>>> escape(''' ' is not escaped''')
" ' is not escaped"

and is it usefull since l89 you already escape it ?

Reply:
I'm just reproducing what was done line 60 (alt = escape(record[options['alt-field']])) I assumed it was correct. If escape is not safe, it needs to be adapted above too.